jaeger-core/src/main/java/io/jaegertracing/Configuration.java

```diff
@@ -186,11 +186,15 @@ public Configuration(String serviceName) {
    * @return Configuration object from environmental variables
    */
   public static Configuration fromEnv() {
-    return new Configuration(getProperty(JAEGER_SERVICE_NAME))
-        .withTracerTags(tracerTagsFromEnv())
-        .withReporter(ReporterConfiguration.fromEnv())
-        .withSampler(SamplerConfiguration.fromEnv())
-        .withCodec(CodecConfiguration.fromEnv());
+    return Configuration.fromEnv(getProperty(JAEGER_SERVICE_NAME));
+  }
+
+  public static Configuration fromEnv(String serviceName) {
```

Comment:
I think it's ok to add this if it is the only init-time required parameter. Otherwise I think the error handling should be moved to `build` method.

Reply:
It's the only required parameter.